plugins/playstation/src/playstation-plugin.vala

```diff
@@ -0,0 +1,38 @@
+// This file is part of GNOME Games. License: GPLv3
+
+private class Games.PlayStation : Object, Plugin {
+	private const string FINGERPRINT_PREFIX = "playstation";
+	private const string MIME_TYPE = "application/x-cue";
+	private const string MODULE_BASENAME = "libretro-playstation.so";
+	private const bool SUPPORTS_SNAPSHOTTING = true;
+
+	public GameSource get_game_source () throws Error {
+		var query = new MimeTypeTrackerQuery (MIME_TYPE, game_for_uri);
+		var connection = Tracker.Sparql.Connection.@get ();
+		var source = new TrackerGameSource (connection);
+		source.add_query (query);
+
+		return source;
+	}
+
+	private static Game game_for_uri (string uri) throws Error {
+		var cue_file = File.new_for_uri (uri);
+		var cue_sheet_handler = new CueSheetHandler (cue_file);
+		var bin_file = cue_sheet_handler.extract_bin_file ();
```

Comment:
We could probably have the cue sheet class to be more generic:
- the sheet could be a list of typed nodes, `FILE` nodes for example
- you could ask for the first `FILE` node
- a `FILE` node could have `TRACK` nodes and a relative path
- a `TRACK` node should have a mode
- we could check the mode is the one we want (not `AUDIO` for example) and extract the relative path to the file

Something like that... what do you think? It would probably be usefull to do it properly now so we can use it as-is with other disc formats.